src/Dev/Deprecation.php

```diff
@@ -234,6 +237,14 @@ public static function notice($atVersion, $string = '', $scope = Deprecation::SC
             } else {
                 user_error($string, self::$notice_level);
             }
+        } catch (Exception $e) {
+            if ($e instanceof BadMethodCallException && $e->getMessage() === InjectorLoader::NO_MANIFESTS_AVAILABLE) {
+                // noop
+                // this can happen when calling Deprecation::notice() before manifests are available, i.e.
+                // some of the code involved in creating the manifests calls Deprecation::notice()
+            } else {
+                throw $e;
```

Comment:
Not sure about this, but I'm assuming that every exception rethrown here, will appear to be coming from `Deprecation::notice()`.

Reply:
I don't believe so, I think it just throws the original `$e` unmodified

```php
php -r '
  class A {
    function f() {
      throw new \Exception();
    }
  }
  class B {
    function f() {
      try {
        (new A())->f();
      } catch (\Exception $e) {
        throw $e;
      }
    }
  }
  class C {
    function f() {
      try {
        (new B())->f();
      } catch (\Exception $e) {
        var_dump($e->getLine());
        var_dump($e->getTraceAsString());
      }
    }
  }
  (new C)->f();
'
int(4)
string(108) "#0 Command line code(10): A->f()
#1 Command line code(19): B->f()
#2 Command line code(26): C->f()
#3 {main}"
```